### PR TITLE
Testability statements for Section 5.1: BPI Subject Capabilities

### PR DESCRIPTION
**Pull request by Therecanbeonlyone1969**

### CODE REVIEW

File: core/baseline-core-v1.0-psd01.md [hunk 1]
```diff
@@ -1287,19 +1470,206 @@ An external BPI Subject MUST at least be able to create, read, update and delete
...
+    - Read the details of the created BPI component and validate that the information is accurate.
+    - Update the BPI component by modifying its attributes (e.g., name, description).
+    - Confirm that the changes to the BPI component are successfully updated and reflected in the system.
+    - Delete the BPI component.
+    - Verify that the BPI Transaction is no longer present in the system.
+
+Passing Criteria:
+* All test passing criteria from [R73](#r73) Testability for an internal BPI subject.
+* For any other BPI component type in the BPI:
+    * The internal BPI Subject can create a new BPI components.
```
Comment (Kasshern) [inline core/baseline-core-v1.0-psd01.md:1573]: component

**Review by Kasshern — COMMENTED** [review]
@Therecanbeonlyone1969  Looks good overall left some comments - mostly spelling and grammar, some suggestions of repeat steps and possibly one or two confusing wordings. Will approve after response.

File: core/baseline-core-v1.0-psd01.md [hunk 2]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
+
+1. BPI Subject encrypts a sample plaintext using a supported cryptographic library.
+2. Verify that the ciphertext produced by the encryption operation is of the expected length and format.
+3. BPI Subject decrypts the ciphertext using the cryptographic library used for encryption.
+4. Confirm that the decrypted plaintext matches the original input.
+5. BPI Subject encrypts a different plaintext using the same cryptographic library but with a different key.
+6. Ensure that the resulting ciphertext is distinct from the previous encryption operation.
+7. BPI Subject decrypts the new ciphertext using the corresponding decryption key.
+8. Validate that the decrypted plaintext accurately represents the original input.
+9. Perform encryption and decryption operations using different cryptographic algorithms supported by the BPI.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md:1316]: no. that is up to the implementer @Kasshern

File: core/baseline-core-v1.0-psd01.md [hunk 3]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
 #### **[R71]**	
-A BPI Subject MUST be capable of performing cryptographic multi-digital-signature creation and verification based on BPI specified cryptographic algorithms.
+A BPI Subject MUST be capable of performing cryptographic threshold-digital-signature creation and verification based on BPI specified cryptographic algorithms.
+
+[R71](#r71) Testability:
+
+Preconditions:
+
+* Set up a BPI system with cryptographic algorithms and protocols.
+* Ensure that BPI Subjects have access to the necessary cryptographic functions for threshold digital signature creation and verification.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md:1400]: no. not required

File: core/baseline-core-v1.0-psd01.md [hunk 4]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
+* Establish a group of participants (BPI Subjects) who will be involved in the threshold digital signature scheme.
+* Establish a digital signature threshold condition.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md:1402]: yes. you need to specify what the number of shares are to reconstitute ahead of time

File: core/baseline-core-v1.0-psd01.md (hunk 1 above)
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md]: done

File: core/baseline-core-v1.0-psd01.md [hunk 5]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
+
+1. Generate a threshold key pair for digital signature for each participant.
+2. Create a sample message or data to be signed.
+3. Perform a threshold digital signature creation process with a subset of participants.
+4. Combine the partial signatures to generate the complete threshold digital signature.
+5. Verify the threshold digital signature using the combined public keys of the participating subjects.
+6. Modify a single character in the signed message and attempt to verify the threshold digital signature.
+7. Confirm that the verification fails, indicating the integrity of the message has been compromised.
+8. Add or remove participants from the group and repeat the threshold digital signature creation and verification process.
+9. Utilize different cryptographic algorithms supported by the BPI for threshold digital signature creation.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md]: agreed and corrected

File: core/baseline-core-v1.0-psd01.md [hunk 6]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
+[R70](#r70) Testability:
+
+Preconditions:
+
+* Set up a BPI system with cryptographic algorithms and protocols.
+* Ensure that the BPI Subject has access to the necessary cryptographic functions for digital signature creation and verification.
+
+Test Steps:
+
+1. BPI Subject generates a key pair for digital signature using the BPI Subject.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md]: corrected

File: core/baseline-core-v1.0-psd01.md [hunk 7]
```diff
@@ -1261,23 +1261,206 @@ A BPI Subject MUST be capable of performing all of the following cryptographic k
...
+* The backup keys can be successfully recovered and restored.
+
 #### **[R68]**	
 A BPI Subject MUST be capable of performing encryption and decryption operations based on BPI specified cryptographic algorithms.
 
+[R68](#r68) Testability:
+
+Preconditions:
+
+* Set up a BPI yesy system with the necessary cryptographic algorithms and protocols.
```
[the remark this answers is not in the document]
Comment (Therecanbeonlyone1969) [reply core/baseline-core-v1.0-psd01.md]: yes

**Review by Ybittan — CHANGES_REQUESTED** [review]

**Review by Ybittan — APPROVED** [review]
LGTM

**Review by Kasshern — APPROVED** [review]

**Pull request merged**